lib/puppet/resource_api/property.rb

```diff
@@ -11,12 +11,29 @@ class Puppet::ResourceApi::Property < Puppet::Property
   # @param attribute_name the name of attribue of the property
   # @param resource_hash the resource hash instance which is passed to the
   # parent class.
-  def initialize(type_name, data_type, attribute_name, resource_hash)
+  def initialize(type_name, data_type, attribute_name, resource_hash, referrable_type)
```

Comment:
Same question as above, but for `Puppet::ResourceApi::Property.new`?

Reply:
```suggestion
  def initialize(type_name, data_type, attribute_name, resource_hash, referrable_type = nil)
```
also requires changes below to assume that the type is not `custom_insync`-able if no referrable_type is passed.